modules/Router.js

```diff
@@ -82,7 +84,10 @@ const Router = React.createClass({
       history,
       createRoutes(routes || children)
     )
+
     const router = createRouterObject(history, transitionManager)
+    router.location = null
+    router.params = null
```

Comment:
I think we were imagining just creating a new `this.router` in `componentWillUpdate` every time, rather than keeping around a persistent one.